chia/server/server.py

```diff
@@ -267,13 +268,29 @@ async def start_server(self, on_connect: Callable = None):
         # this port from the socket itself and update self._port.
         self.site = web.TCPSite(
             self.runner,
-            host="0.0.0.0",
+            host="",  # should listen to both IPv4 and IPv6 on a dual-stack system
             port=int(self._port),
             shutdown_timeout=3,
             ssl_context=ssl_context,
         )
         await self.site.start()
-        self._port = self.runner.addresses[0][1]
+        #
+        # On a dual-stack system, we want to get the (first) IPv4 port unless
+        # prefer_ipv6 is set in which case we use the IPv6 port
+        #
+        if self._port == 0:
+            global_config = load_config(self.root_path, "config.yaml")
+            prefer_ipv6 = global_config.get("prefer_ipv6", False)
+            self._port = self.runner.addresses[0][1]  # sets the default to handle some edge cases
+            for x in self.runner.addresses:
+                ip_addy = ip_address(x[0])
```

Comment:
```suggestion
                address = ip_address(raw_address)
```

Reply:
thanks, used this to make the code more readable